cg/models/pipeline_qc_metrics/pipeline_qc_metrics.py

```diff
@@ -0,0 +1,113 @@
+from datetime import datetime
+
+from pydantic import BaseModel
+
+
+class LaneMetric(BaseModel):
+    """Model for lane specific sequencing metrics."""
+
+    lane: int
+    number_of_reads: int
+    _yield: int | None = None
+
+
+class ConcatenatedLaneMetric(BaseModel):
+    """Model for concatenated lane sequencing metrics."""
+
+    number_of_reads: int
+    _yield: int | None = None
+
+
+class DNACoverageMetric(BaseModel):
+    """Coverage metrics from the pipeline output."""
+
+    mean_target_coverage: float | None = None
+    median_target_coverage: float | None = None
+    pct_target_bases_10x: float | None = None
+    pct_target_bases_20x: float | None = None
+    pct_target_bases_30x: float | None = None
+    pct_target_bases_40x: float | None = None
+    pct_target_bases_50x: float | None = None
+    pct_target_bases_100x: float | None = None
+    pct_target_bases_250x: float | None = None
+    pct_target_bases_500x: float | None = None
+    pct_target_bases_1000x: float | None = None
+    gc_dropout: float | None = None
+    at_dropout: float | None = None
+    pct_off_bait: float | None = None
+    pct_pf_reads_improper_pairs: float | None = None
+    pct_chimeras: float | None = None
+
+
+class RNACoverageMetric(BaseModel):
+    """Coverage metrics for an RNA pipeline output."""
+
+    after_filtering_gc_content: float | None
+    after_filtering_q20_rate: float | None
+    after_filtering_q30_rate: float | None
+    after_filtering_read1_mean_length: float | None
+    median_5prime_to_3prime_bias: float | None
+    pct_adapter: float | None
+    pct_mrna_bases: float | None
+    pct_ribosomal_bases: float | None
+    pct_surviving: float | None
+    pct_duplication: float | None
+    read_pairs_examined: float | None
+    uniquely_mapped_percent: float | None
+
+
+class SequencingMetric(BaseModel):
+    """Model for collecting sample specific reads information."""
+
+    total_reads: list[LaneMetric] | ConcatenatedLaneMetric
+    mapped_reads: list[LaneMetric] | ConcatenatedLaneMetric
+    percentage_mapped_reads: list[float] | float
+    coverage_metric: DNACoverageMetric | RNACoverageMetric
+    pct_duplication: float | None = None
+    pct_adapter: float | None
+
+
+class MIPDNAMetric(BaseModel):
+    sex: str
+
+
+class MIPRNAMetric(BaseModel):
+    pass
+
+
+class BalsamicMetric(BaseModel):
+    pass
```

Comment:
I don't know if this is all intended to be for QC trending, or if we could also add more sample-specific / analysis specific metrics, like:
- tumor-purity.
- TMB-score
- MSI status
 I suppose if we add as much information as possible this model could be used to extract information to create delivery-reports with all sorts of interesting sample-specific info, like

Reply:
Yes, great that you mention these. It would be great if @mathiasbio and @ivadym  could make a list of such metrics so we can add them to the model here!